addons/csi-azuredisk/ccm-azuredisk.yaml

```diff
@@ -657,3 +691,828 @@ status:
+              name: azure-cred
+          resources:
+            limits:
+              memory: 500Mi
+            requests:
+              cpu: 10m
+              memory: 20Mi
+      volumes:
+        - name: socket-dir
+          emptyDir: {}
+        - name: azure-cred
+          hostPath:
+            path: /etc/kubernetes/
+            type: DirectoryOrCreate
+---
+# Source: azuredisk-csi-driver/templates/csi-snapshot-controller.yaml
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: csi-snapshot-controller
+  namespace: kube-system
+  labels:
+    app.kubernetes.io/instance: "release-name"
+    app.kubernetes.io/managed-by: "Helm"
+    app.kubernetes.io/name: "azuredisk-csi-driver"
+    app.kubernetes.io/version: "v1.27.1"
+    helm.sh/chart: "azuredisk-csi-driver-v1.27.1"
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: csi-snapshot-controller
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/instance: "release-name"
+        app.kubernetes.io/managed-by: "Helm"
+        app.kubernetes.io/name: "azuredisk-csi-driver"
+        app.kubernetes.io/version: "v1.27.1"
+        helm.sh/chart: "azuredisk-csi-driver-v1.27.1"
+        app: csi-snapshot-controller
+    spec:
+      serviceAccountName: csi-snapshot-controller-sa
+      nodeSelector:
+        kubernetes.io/os: linux
+      priorityClassName: system-cluster-critical
+      tolerations:
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/master
+          operator: Exists
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/controlplane
+          operator: Exists
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/control-plane
+          operator: Exists
+      containers:
+        - name: csi-snapshot-controller
+          image: "{{ .InternalImages.Get "AzureDiskCSISnapshotterController" }}"
+          args:
+            - "--v=2"
+            - "--leader-election=true"
+            - "--leader-election-namespace=kube-system"
+          resources:
+            limits:
+              memory: 100Mi
+            requests:
+              cpu: 10m
+              memory: 20Mi
+          imagePullPolicy: IfNotPresent
+---
+# Source: azuredisk-csi-driver/templates/csi-azuredisk-driver.yaml
+apiVersion: storage.k8s.io/v1
+kind: CSIDriver
+metadata:
+  name: disk.csi.azure.com
+  annotations:
+    csiDriver: "v1.27.1"
+    snapshot: "v5.0.1"
+spec:
+  attachRequired: true
+  podInfoOnMount: false
+  fsGroupPolicy: File
+---
+# Source: azuredisk-csi-driver/templates/csi-snapshot-controller.yaml
+kind: VolumeSnapshotClass
+apiVersion: snapshot.storage.k8s.io/v1
+metadata:
+  name: csi-azuredisk-vsc
+  annotations:
+    # This is what defines this resource as a hook. Without this line, the
+    # job is considered part of the release.
+    "helm.sh/hook": post-install
+driver: disk.csi.azure.com
+deletionPolicy: Delete
+parameters:
+  incremental: "true"
+  tags:
```

Comment:
This should be part of default-storage-class addon

Reply:
Are you sure? It's no a StorageClass, it's VolumeSnapshotClass.

Reply:
Yes. We also keep VolumeSnapshotClasses for other providers in that addon.